nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -217,22 +232,29 @@ class NodeletLazy: public nodelet::Nodelet
     */
   template<class T> ros::Publisher
   advertise(ros::NodeHandle& nh,
-            std::string topic, int queue_size, bool latch=false)
+            const std::string& topic, int queue_size, bool latch=false)
   {
     boost::mutex::scoped_lock lock(connection_mutex_);
     ros::SubscriberStatusCallback connect_cb
-      = boost::bind(&NodeletLazy::connectionCallback, this, _1);
-    ros::SubscriberStatusCallback disconnect_cb
-      = boost::bind(&NodeletLazy::connectionCallback, this, _1);
+      = boost::bind(&NodeletLazy::connectionCallback, this);
     ros::Publisher pub = nh.advertise<T>(topic, queue_size,
                                           connect_cb,
-                                          disconnect_cb,
+                                          connect_cb,
                                           ros::VoidConstPtr(),
                                           latch);
     publishers_.push_back(pub);
     return pub;
   }
 
+  /**
+   * @brief Checkes if at least one publisher has subscribers.
+   * @return True if at least one publisher has subscribers; False otherwise.
+   */
+  virtual bool hasSubscribers() const
```

Comment:
A child class can overload this to support other type of publishers with something like this: `return NodeletLazy::hasSubscribers() || nodelet_topic_tools::has_subscribers(image_transport_publishers_);`

@dirk-thomas Any idea where can I contribute the child class that I have to support both `ros::Publisher` and `image_transport::Publisher`? For now I have it on a private repo.
Maybe `image_transport`, but then it'd have to depend on `nodelet_topic_tools`, which doesn't sound good.

Reply:
> Maybe `image_transport`

Sounds like the best option to me.

Reply:
@vrabaud  Could you comment on this?

Long story short, I've done some changes here so it's possible to extend the `nodelet_topic_tools::NodeLazy` class to support both `ros::Publisher` and `image_transport::Publisher`.

Now I'm looking for a repo where I can contribute the derived class that provides that functionality. If you think `image_transport` is a good place, where in there could it be? Maybe a new package, b/c this will need a dependency on `nodelet_topic_tools`.